### ISSUE TITLE
Load Keyframes from Motion Tracker sometimes doesn't work

### ISSUE THREAD

**Issue by ethicnology** [mltframework/shotcut#1472]
## Context
os:       `Apple M2`  `macOS ventura 13.5.2` 
shotcut: `23.07.29` (`brew`) & `23.09.12` (`.dmg`)

## Problem:
**Load Keyframes from Motion Tracker** button **sometimes** doesn't work.

## Scenario:
- Open a fresh shotcut with a video
- Use the motion tracker filter on this video and press **analyze**
- Add simple mask or crop rectangle
- Push the button [**Load Keyframes from Motion Tracker**]
- The keyframes are not loaded

## Temporary solution:
- Save project
- Exit shotcut
- Start shotcut again

**Comment by ddennedy** [comment]
I did not reproduce it using those exact steps. Please explain more "Open a fresh shotcut with a video." For that step I did not use Playlist or Timeline. Did you? I just tried adding to timeline and using it from there, and it worked too. When you write "sometimes" are you saying that it is intermittent when repeating the steps above or do you mean there might be other steps needed that you have not figured out?
When you write "keyframes are not loaded" do you mean you are able to choose options in the dialog and click Apply but no keyframes are generated? Or do you mean the tracker is not listed in the dialog?

**Comment by Stasof** [comment]
I have the same problems, the track is invisible in the dialog or then when I press the Apply button, nothing happens. I was able to change the position of a video (chroma key over main video) using a motion tracker, only on a simple video, in a project with a lot of videos, in 99% of cases the function does not work.

**Comment by martinschnurer** [comment]
Had the same problem. When clicked on "Load keyframes from Motion Tracker", then confirmed modal and "Apply" nothing happened. After restarting Shotcut it finally worked (it loaded after clicking on Apply).

**Comment by khinch** [comment]
Same problem here. Version 24.04.28 on NixOS. The keyframes aren't exported until shotcut is closed and reopened. It will let me _try_ to load them for another filter without errors, but they don't get applied.

**Comment by busyBeaver1** [comment]
Had a similar problem with a big project. Also worked after reopening Shotcut. Version 24.06 on Arch

**Comment by ethicnology** [comment]
`stale`

**Comment by lukas-reinert-datascience** [comment]
Got the same issue you describe. Yet I may have noticed a pattern that might help with fixing the issue. 
I am only unable to load keyframes, if I add a new clip to the timeline after I ran the motion tracking analysis. 

However, if I add all the clips I want to apply motion tracker data to first, I can run as many motion tracking analyses/apply the keyframes as I want to.

In any case, re-starting shotcut also fixes the issue for the next analyses/keyframe application. It's not untill I add a new clip to my project, that the keyfram application stops working.



In summary:
### What works:
Adding all the clips you want to perform motion tracking on -> Running analyses -> Applying all keyframes

### Issue comes up when:
Adding clip 1 -> Running analysis 1 -> Applying key frames 1 (works) -> Adding clip 2 -> Running Analysis 2 -> Applying key frames 2 (does not work)

_________________________________________________________________________
System spec: 
Windows 10
Shotcut 24.09.13

**Comment by ddennedy** [comment]
> I am only unable to load keyframes, if I add a new clip to the timeline

This prompted me to investigate. I reproduced the problem, but this is not quite correct. If you drag a file from a file manager, **Playlist**, or **Files** to the **Timeline** the problem does not occur. The problem actually occurs whenever you open something into the **Source** player: **File > Open**, **Recent**, double-click **Playlist** item, or double-click file in **Files**.
I have a fix for the next version.
